### PR TITLE
fix innerHTML XSS vulnerability

### PR DESCRIPTION
**Pull request by drewe**

### CODE REVIEW

File: io-string.html [hunk 1]
```diff
@@ -36,24 +42,21 @@
...
-          this.immediateValue = this.$.value.innerHTML;
+          this.value = this.immediateValue;
         },
         immediateValueChanged: function() {
           if (this.updateImmediate) {
-            this.value = this.immediateValue;
-          }
-          if (!this.focused) {
-            this.$.value.innerHTML = this.immediateValue != undefined ? this.immediateValue : '';
+            this.updateImmediateValue();
```
Comment (arodic) [inline io-string.html]: Would this always update the value, even if updateImediate is false? The idea was to have the value update only when you finish editing and update immediateValue on each keystroke.

Comment (drewe) [inline io-string.html]: right. So we call updateImmediateValue if updateImmediate is true, and on blur.

**Comment by arodic** [comment]
Merged. Thanks!

**Comment by arodic** [comment]
New tag v0.9.9

**Pull request merged**